File: backend/config_manager.py
"""
Configuration management for PiRouter
"""
import json
from pathlib import Path
from models import RouterConfig

# Configuration paths
WIREGUARD_DIR = Path("/etc/wireguard")
WG_INTERFACE = "wg0"  # Single WireGuard interface name
WG_CONFIG_FILE = WIREGUARD_DIR / f"{WG_INTERFACE}.conf"

# VPN profiles directory (stored separately from active config)
USER_HOME = Path("/home/pgj99")  # Explicit path since service runs as root

if Path("/app").exists():
    VPN_PROFILES_DIR = Path("/app/config/vpn_profiles")
    CONFIG_FILE = Path("/app/config/router_config.json")
else:
    VPN_PROFILES_DIR = USER_HOME / "code/PiRouter/backend/config/vpn_profiles"
    CONFIG_FILE = USER_HOME / "code/PiRouter/backend/config/router_config.json"

BYPASS_TABLE = "100"

# Ensure VPN profiles directory exists
VPN_PROFILES_DIR.mkdir(parents=True, exist_ok=True)


def load_config() -> RouterConfig:
    """Load router configuration from JSON file"""
    if CONFIG_FILE.exists():
        with open(CONFIG_FILE, 'r') as f:
            data = json.load(f)
            return RouterConfig(**data)
    return RouterConfig()


def save_config(config: RouterConfig):
    """Save router configuration to JSON file"""
    CONFIG_FILE.parent.mkdir(parents=True, exist_ok=True)
    with open(CONFIG_FILE, 'w') as f:
        json.dump(config.dict(), f, indent=2)

File: PiRouterBackend/Services/ConfigManager.cs
using System.Text.Json;
using PiRouterBackend.Models;

namespace PiRouterBackend.Services;

public interface IConfigManager
{
    RouterConfig LoadConfig();
    Task SaveConfigAsync(RouterConfig config);
    string GetVpnProfilesDir();
    string GetConfigFilePath();
    string GetWireGuardDir();
}

public class ConfigManager : IConfigManager
{
    private readonly string _vpnProfilesDir;
    private readonly string _configFile;
    private readonly string _wireGuardDir;
    private readonly ILogger<ConfigManager> _logger;
    private readonly string _userHome;

    public ConfigManager(ILogger<ConfigManager> logger)
    {
        _logger = logger;
        _userHome = Environment.GetEnvironmentVariable("HOME") ?? "/home/pgj99";

        if (Directory.Exists("/app"))
        {
            _vpnProfilesDir = "/app/config/vpn_profiles";
            _configFile = "/app/config/router_config.json";
        }
        else
        {
            _vpnProfilesDir = Path.Combine(_userHome, "code/PiRouter/wireguard_configs");
            _configFile = Path.Combine(_userHome, "code/PiRouter/backend/config/router_config.json");
        }

        _wireGuardDir = "/etc/wireguard";

        // Ensure directories exist
        Directory.CreateDirectory(Path.GetDirectoryName(_vpnProfilesDir)!);
        Directory.CreateDirectory(Path.GetDirectoryName(_configFile)!);
    }

    public RouterConfig LoadConfig()
    {
        try
        {
            if (File.Exists(_configFile))
            {
                var json = File.ReadAllText(_configFile);
                var config = JsonSerializer.Deserialize<RouterConfig>(json);
                return config ?? new RouterConfig();
            }
        }
        catch (Exception ex)
        {
            _logger.LogError(ex, "Error loading config");
        }

        return new RouterConfig();
    }

    public async Task SaveConfigAsync(RouterConfig config)
    {
        try
        {
            var options = new JsonSerializerOptions { WriteIndented = true };
            var json = JsonSerializer.Serialize(config, options);
            await File.WriteAllTextAsync(_configFile, json);
        }
        catch (Exception ex)
        {
            _logger.LogError(ex, "Error saving config");
            throw;
        }
    }

    public string GetVpnProfilesDir() => _vpnProfilesDir;
    public string GetConfigFilePath() => _configFile;
    public string GetWireGuardDir() => _wireGuardDir;
}
